### DISCUSSION TITLE
Flow direction on interface

### DISCUSSION THREAD

**Discussion by freedomwarrior** [akvorado/akvorado#431] [Q&A]
Hello.
I'm using Juniper MX series and when I config flow on interfaces I see that I can enable it on both direction input and output.
But in documentation there is example only with input direction on interface.
So, do I need enable it on both directions or only one(input) ?
Thank you.

**Comment by vincentbernat** [comment]
You have two solutions:
1. Enable on all IP interfaces in ingress (compatible with most routers)
2. Enable only on external interfaces in both ingress/egress (this may or may not work depending on the router)

Don't mix, otherwise you will have duplicate flows. One flow should be captured only once.

**Comment by ghost** [comment]
Hello,
again, thanks for akvorado!

Yesterday I switched from option 1. with sampling ingress on edge and back-bone interfaces to option 2. with sampling ingress and egress on all edge interfaces.

I am wondering what the correct filters are to not have "wrongly" counted flows?

![akvorado-traffic](https://github.com/akvorado/akvorado/assets/87298913/87365222-a438-448c-8c46-20e959d74e75)

The steep increase around 10:00 o'clock is the change of the flow sampling. With the "previous period" (great feature btw!) the diff is clearly visible. Using `InIfBoundary = external AND OutIfBoundary = internal` as a filter show more correct traffic levels, but is obviously not correct  as I am missing the `OuIfBoundary = external` traffic.

Probabaly the solution is quite simple and obvious, but I am missing a clue atm.

thanks a lot and sorry for bothering!
